kong/db/schema/entities/cas.lua

```diff
@@ -0,0 +1,41 @@
+local typedefs = require "kong.db.schema.typedefs"
+local openssl_x509 = require "openssl.x509"
+local ngx_time = ngx.time
+
+return {
+  name        = "cas",
+  primary_key = { "id" },
+
+  fields = {
+    { id             = typedefs.uuid, },
+    { created_at     = typedefs.auto_timestamp_s },
+    { cert           = typedefs.certificate { required = true }, },
+    { tags           = typedefs.tags },
+  },
+
+  entity_checks = {
+    { custom_entity_check = {
+      field_sources = { "cert", },
+      fn = function(entity)
+        local cert = openssl_x509.new(entity.cert)
+        local not_before, not_after = cert:getLifetime()
+        local now = ngx_time()
+
+        if not_before > now then
```

Comment:
is this a desired constraint? isn't it a case that someone may be preparing a certificate so it kicks in at some point in the future? (I assume that's the use of the `Not Before` attribute). In that case I don't see a problem in storing it in the database.